MdePkg/Library/StackCheckLib/AArch64/DynamicCookieGcc.S

```diff
@@ -0,0 +1,78 @@
+#------------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+# Module Name:
+#
+#   DynamicCookie.S
+#
+# Abstract:
+#
+#   Generates random number through the RNDR instruction on a 64-bit AARCH64 platform
+#   to store a random value in the GCC __stack_check_guard stack cookie.
+#   The first byte is 0'd to prevent string copy functions from clobbering
+#   the stack cookie.
+#
+# Notes:
+#
+# If RNDR fails, the build time static stack cookie value will be used instead.
+#
+#------------------------------------------------------------------------------
+
+.text
+.p2align 2
+
+GCC_ASM_IMPORT(__stack_chk_guard)
+GCC_ASM_IMPORT(_CModuleEntryPoint)
+GCC_ASM_EXPORT(_ModuleEntryPoint)
+
+#define RNDR  S3_3_C2_C4_0
+
+#------------------------------------------------------------------------------
+# VOID
+# EFIAPI
+# _ModuleEntryPoint (
+#   Parameters are passed through.
+# )
+#------------------------------------------------------------------------------
+ASM_PFX(_ModuleEntryPoint):
+  AARCH64_BTI(c)
+  stp x29, x30, [sp, #-16]!         // Save frame pointer and link register
+  stp x19, x20, [sp, #-16]!         // Save x19 and x20
+  stp x21, x22, [sp, #-16]!         // Save x21 and x22
+  stp x23, x24, [sp, #-16]!         // Save x23 and x24
+  stp x25, x26, [sp, #-16]!         // Save x25 and x26
+  stp x27, x28, [sp, #-16]!         // Save x27 and x28
+  stp x0, x1, [sp, #-16]!           // Save x0 and x1
+  stp x2, x3, [sp, #-16]!           // Save x2 and x3
+  stp x4, x5, [sp, #-16]!           // Save x4 and x5
+  stp x6, x7, [sp, #-16]!           // Save x6 and x7
+

```

Comment:
What's all this for?

Reply:
I think in my initial writing of this, I had clobbered some registers and broke the entry point, so then I went super aggressive in saving everything. I can trim this down to only save the registers that I use.

Reply:
Yeah, this evolved a bit and I'm not sure why I used `x0` and `x1` :). I also need to avoid `x2` and `x3` for the StMM ARM64 entry point: https://github.com/tianocore/edk2/blob/8b87eb9dfba054331ed16204f36c4885aefc3c94/StandaloneMmPkg/Library/StandaloneMmCoreEntryPoint/Arm/StandaloneMmCoreEntryPoint.c#L332-L336.

But I'll update this to use other registers.